Add struct for ppc64le architecture

krishvoor opened the pull request:
This PR adds struct for ppc64le architecture,
`lau-jit.arch()` reads PowerPC-Little Endian Architecture as ppc64le,
corresponding data types change to the `typedef struct`

krishvoor commented:
FYI/A - @afsanjar @andrewdavidmackenzie ^^^
With these changes -
```
bash-4.4$ apicast 
env: '/tmp/JWw6lchEDo': Permission denied
loading production environment configuration: /opt/app-root/src/config/production.lua
nginx: [crit] ngx_slab_alloc() failed: no memory
bash-4.4$ 
bash-4.4$ arch
ppc64le
bash-4.4$ 
bash-4.4$ luajit 
LuaJIT 2.1.0-beta3 -- Copyright (C) 2005-2020 Mike Pall. https://luajit.org/
JIT: OFF
> print (jit.arch)
ppc64le
> 
```

afsanjar commented:
+1

unleashed commented on lfs_ffi.lua:
```diff
@@ -801,33 +801,33 @@ if OS == 'Linux' then
             stat_syscall_num = 195
             lstat_syscall_num = 196
         end
-    elseif ARCH == 'ppc' or ARCH == 'ppcspe' then
```
Why remove this one? Is it causing issues building for `ppc64le`?

afsanjar replied:
PPC is returned on some P6 AIX systems (big-endian) still in service with some banks and government agencies in the US.
ppcscpe is code for old PS3s based on IBM Cell Processor.

afsanjar replied:
ppc64le is a different processor.
Agreed,  'ppc' and 'ppcspe' should not be removed.

krishvoor replied:
Agreed, did not touch `ppc` & `ppcspe`

krishvoor replied:
PR now only adds new `ARCH = ppc64le` and corresponding data types.

unleashed commented on lfs_ffi.lua:
```diff
@@ -801,33 +801,33 @@ if OS == 'Linux' then
                 long long       st_size;
-                int             st_blksize;
-                int             __pad2;
+                long long       st_blksize;
```
Please use `unsigned long` and `long` instead of `unsigned long long` and `long long` on 64 bit arches. Ref: https://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.html#FUND-TYPE.

krishvoor replied:
Ok, I have modified them as per https://refspecs.linuxfoundation.org/ELF/ppc64/PPC-elf64abi.html#FUND-TYPE

unleashed reviewed (changes requested):
Hi @krishvoor, the patch looks good except for a few nits I'd like addressed:

1. Does the `ppc`/`ppcspe` arch present in upstream bother? If not, I'd like to keep it to limit the patch to a minimum.
2. Please don't use `long long` or `unsigned long long`, as this is very confusing to someone reading C types for a 64 bit CPU, since they should match `long` and `unsigned long`.

Can you also squash the changes in a single commit?

Commit:
Add struct for ppc64le architecture

This PR adds struct for ppc64le architecture,
`lau-jit.arch()` reads PowerPC-Little Endian Architecture as ppc64le,
corresponding data types change to the `typedef struct`

Signed-off-by: Krishna Harsha Voora <krishvoor@in.ibm.com>

unleashed commented on lfs_ffi.lua:
```diff
@@ -801,6 +801,33 @@ if OS == 'Linux' then
+                unsigned int    st_uid;
+                unsigned int    st_gid;
+                int             __pad2;
+                unsigned long long st_rdev;
```
@krishvoor forgot about this entry

krishvoor replied:
Good catch, fixed now!

unleashed reviewed (approved):
Looks good, thanks @krishvoor!
